### PR TITLE
Add nightly feature gate for const

### LINKED ISSUE #8: provide const_trait_impls behind a gate
It would be nice of you if you could provide all the trait impls as const.
Currently afaik all the bitops, maybe from impls and the Number trait would be useful as const.
Not entirely sure how easy it is to put this behind a feature gate or something like that.
Otherwise I have to vendor somehow or just work without const for a while.
Thank you for your time.

### PR DESCRIPTION
**Pull request by hecatia-elegua**
Closes #8.

It has been a while, but now I cleaned this up finally.

### CODE REVIEW

**Comment by hecatia-elegua** [comment]
Oh and could you please publish a new version? Thanks

**Comment by hecatia-elegua** [comment]
@danlehmann since I want to publish a crate depending on this functionality soon, could we please move forward with this?

File: Cargo.toml [hunk 1]
```diff
@@ -12,6 +12,7 @@ categories = ["embedded", "no-std", "data-structures"]
 
 [features]
 std = []
+nightly = []
```
Comment (danlehmann) [inline Cargo.toml:15]: I have not published a crate that exposes an optional nightly feature before.

What are the trade-offs of calling this "nightly" vs "nightly_const_convert"/"nightly_const_trait_impl"?

**Review by danlehmann — COMMENTED** [review]
Thanks for your contribution (and sorry again for my late response!)

One quick question, then I'll be happy to merge this

File: Cargo.toml (hunk 1 above)
Comment (hecatia-elegua) [reply Cargo.toml:15]: I suppose if someone just wants one of them, they could file an issue and we could still do `nightly = [nightly_const_convert, nightly_const_trait_impl]`.
I remember some project did that, but not which one.

Besides that, if we had those separated, if one of them becomes stable we could... not do anything with that info either. Maybe someone really only needs Number to be a const trait, but we could provide that if someone asks. The other one, const_convert, needs const trait iirc.

If we add more nightly functionality, it's either needed, so we add it to the nightly feature, or special, which would need a different feature.

Comment (hecatia-elegua) [reply Cargo.toml:15]: Ok, found a funny way to contact you, here, in this comment I will delete right after:
are you on discord or sth? I'm only on discord and... mail.

**Comment by hecatia-elegua** [comment]
take two: do you get notified about this?
could you add me on discord?
`hecatia-elegua#1559`

**Comment by danlehmann** [comment]
Yes, just sent you a request. Feel free to delete the message

**Pull request merged**